### PR TITLE
Enable GCP native fencing

### PR DESCRIPTION
**Pull request by BillAnastasiadis**
This PR enables GCP native fencing, by passing the required variables from terraform to ansible and adding the gcp-specific stonith commands for ansible.

- Related ticket: https://jira.suse.com/browse/TEAM-8704

# Verification Run

### Native fencing
-  http://openqaworker15.qa.suse.cz/tests/296261 :green_circle:  --> http://openqaworker15.qa.suse.cz/tests/296261/logfile?filename=deploy-ansible.sap-hana-cluster.log.txt
-  http://openqaworker15.qa.suse.cz/tests/296350  :green_circle: 

### SBD
GCP SBD: http://openqaworker15.qa.suse.cz/tests/296342

## AWS
AWS Native: http://openqaworker15.qa.suse.cz/tests/296341

## AZURE
AZURE MSI: http://openqaworker15.qa.suse.cz/tests/296343

### CODE REVIEW

File: ansible/playbooks/tasks/cluster-bootstrap.yaml [hunk 1]
```diff
@@ -233,6 +233,14 @@
       params tag={{ aws_stonith_tag}} pcmk_delay_max=15
   when: cloud_platform_is_aws
 
+- name: Set primary and secondary hostnames
+  set_fact:
```
Comment (mpagot) [inline ansible/playbooks/tasks/cluster-bootstrap.yaml:237]:
```suggestion
  ansible.builtin.set_fact:
```

Comment (BillAnastasiadis) [reply ansible/playbooks/tasks/cluster-bootstrap.yaml]: Changed!

**Review by alvarocarvajald — APPROVED** [review]
LGTM

File: ansible/playbooks/tasks/cluster-bootstrap.yaml [hunk 2]
```diff
@@ -233,6 +233,14 @@
       params tag={{ aws_stonith_tag}} pcmk_delay_max=15
   when: cloud_platform_is_aws
 
+- name: Set primary and secondary hostnames
+  ansible.builtin.set_fact:
+    primary_hostname: "{{ groups['hana'][0] }}"
```
Comment (mpagot) [inline ansible/playbooks/tasks/cluster-bootstrap.yaml:238]: We use similar query to calculate `is_primary`. Differences are:
- all of them calculate it ad playbook level and not at task level
- other places are using `group.hana[0]`
- other places are using a combination like:
```
hosts: hana
vars:
        is_primary: "{{ ansible_play_hosts[0] == inventory_hostname }}"
```


Let's keep it like this here but revisit it everywhere in all files.

File: terraform/gcp/inventory.tmpl [hunk 3]
```diff
@@ -4,6 +4,9 @@ all:
     use_sbd: ${use_sbd}
     gcp_cluster_ip: ${hana-vip}
     prefix: ${name_prefix}
+    project: ${gcp_project}
+    primary_zone: ${gcp_primary_zone}
```
Comment (mpagot) [inline terraform/gcp/inventory.tmpl:8]: Similar to the other comment, we can think about moving them to hosts>>hana section of the playbook

**Review by mpagot — APPROVED** [review]
LGTM

**Review by lilyeyes — APPROVED** [review]
Generally LGTM.

**Pull request merged**